### PR TITLE
Query Frontend: Add tests and fix previous bugs for instant query tripperware

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

Follow up PR for https://github.com/thanos-io/thanos/pull/5561.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

1. Add the missing changelog.
2. Add more test cases and PromQL compatibility tests for query frontend with sharding.
3. Fix results to handle `string` and `matrix` for instant queries. Those types are not handled correctly in the previous pr.

### CODE REVIEW

File: CHANGELOG.md
Comment (yeya24) [inline CHANGELOG.md:37]: This is added for previous pr.

File: internal/cortex/querier/queryrange/queryrange.proto
Comment (yeya24) [inline internal/cortex/querier/queryrange/queryrange.proto:72]: Handled all 4 cases mentioned in https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries.

**Review by fpetkovski — COMMENTED** [review]
I remember running an instant query with a range selector and getting an error, so something like `http_requests_total[2m]` or `http_requests_total[5m:1m]`. Can we cover this case with a unit test?

**Comment by yeya24** [comment]
Sure, I will add that.

**Comment by yeya24** [comment]
cc @fpetkovski @GiedriusS for another look 😄

**Review by fpetkovski — APPROVED** [review]
Thanks!

**Comment by yeya24** [comment]
Need another approval from maintainers and then we can start cutting the rc.

**Review by GiedriusS — APPROVED** [review]

**Comment by GiedriusS** [comment]
:muscle:

**Pull request merged**
